Mesh_3/include/CGAL/Mesh_3/polylines_to_protect.h

```diff
@@ -301,6 +301,19 @@ struct Polyline_visitor
 template <typename Kernel>
 struct Angle_tester
 {
+  const double m_angle_rad;//in radian to avoid extra computations
+  const bool m_test_angle;
+
+  Angle_tester()
+    : m_angle_rad(0.5 * CGAL_PI)//stored in radian
+    , m_test_angle(false)//under 90 we check for acute, with exact computation
+  {}
+
+  Angle_tester(const double angle_deg)//angle given in degrees for readability
+    : m_angle_rad(angle_deg * CGAL_PI /180.)//stored in radian
+    , m_test_angle(angle_deg != 90.)//under 90 we check for acute, with exact computation
+  {}
```

Comment:
We should document, and ensure, that the angle must be bigger than 90°: if the angle between [p, p1] and [p p2] is smaller than 90°, then Mesh_3 requires that the vertex is a terminal vertex.

Then, I think than, instead of storing the actual angle, we should store the cosine of the angle, to avoid a call to `std::acos` for each call to `operator()`.

Reply:
Actually, as we know that the angle is > 90°, then the sign of the cosine is negative, so let's store the squared cosine directly:
```suggestion
  Angle_tester(const double angle_deg)//angle given in degrees for readability
    : m_angle_sq_cosine(CGAL::square(std::cos((std::min)(90, angle_deg) * CGAL_PI /180.))) // squared cosine of `std:min(90, angle_deg)`
    , m_test_angle(angle_deg != 90.)//under 90 we check for acute, with exact computation
  {}
```